bee-network/bee-autopeering/src/request.rs

```diff
@@ -109,13 +92,14 @@ impl RequestManagerInner {
             response_tx,
         };
 
-        let _ = self.open_requests.insert(key, value);
+        let _ = self.open_requests.write().expect("write").insert(key, value);
```

Comment:
I don't know `RwLock` well, but do we need to catch that error?

Reply:
I think this is okay -- this returns an error if the lock is poisoned, in which case something has gone badly wrong.

Reply:
yup, as Adam says. This only panics if the lock was poisened, and in this case the node is not operating normally anymore and shouldn't do ledger or other critical operations.